platforms/org.eclipse.mita.platform.xdk110/src/org/eclipse/mita/platform/xdk110/platform/Validation.xtend

```diff
@@ -149,28 +133,72 @@ class Validation implements IResourceValidator {
 			}
 		].toSet
 		
-		val i2cReads   = new HashSet(i2cs ) => [retainAll(reads)]
-		val i2cWrites  = new HashSet(i2cs ) => [retainAll(writes)]
-		val gpioReads  = new HashSet(gpios) => [retainAll(reads)]
-		val gpioWrites = new HashSet(gpios) => [retainAll(writes)]
-		
+		val i2cReads   = new HashSet(i2cs  ) => [retainAll(reads)]
+		val i2cWrites  = new HashSet(i2cs  ) => [retainAll(writes)]
+		val gpioReads  = new HashSet(gpios ) => [retainAll(reads)]
+		val gpioWrites = new HashSet(gpios ) => [retainAll(writes)]
+				
 		i2cReads.forEach[validateI2cReadWrite(it.source, it.sigInst, it.structFeature, "Read", "read from", acceptor)]
 		i2cWrites.forEach[validateI2cReadWrite(it.source, it.sigInst, it.structFeature, "Write", "write to", acceptor)]
 		
 		i2cWrites.forEach[validateI2cWriteLength(it.source, it.method, it.sigInst, acceptor)]
 		
 		gpioReads.forEach[validateGpioRead(it, acceptor)]
 		gpioWrites.forEach[validateGpioWrite(it, acceptor)]
+		
+		noises.forEach[validateNoiseRead(noises, it, acceptor)]
```

Comment:
Couldn't this be part of `NoiseSensorValidator`?

Reply:
sure